go-controller/pkg/cni/helper_linux.go

```diff
@@ -450,8 +452,65 @@ func ConfigureOVS(ctx context.Context, namespace, podName, hostIfaceName string,
 		ovsArgs = append(ovsArgs, []string{"--", "--if-exists", "remove", "interface", hostIfaceName, "external_ids", types.NADExternalID}...)
 	}
 
-	if out, err := ovsExec(ovsArgs...); err != nil {
-		return fmt.Errorf("failure in plugging pod interface: %v\n  %q", err, out)
+	// 1. plug in ovs port
+	// 2. do following for dpu mode
+	//   2.1 if hw offload is enabled, verify offload by counting tc ingress filter
+	//   2.2 if offload verification fails, retry up to 3 times every 100 ms
+	hwOffloadEnabled, err := isHWOffloadEnabled()
```

Comment:
@jxiaobin just a nit,would you find it somewhat cleaner to encapsulate the retry in a separate method and have here something like
```
if hwOffloadEnabled {
   addPortArgs := ovsArgs
   delPortArgs := []string{ "--if-exists", "del-port", "br-int", hostIfaceName}
   err := configureOVSHWOffloadWithRetry(ovsArgs, delPortArgs)
   if err != nil {
       return err   
   }
} else {
    out, err := ovsExec(ovsArgs...)
    if err != nil {
        return fmt.Errorf("failure in plugging pod interface: %v\n  %q", err, out)
    }
}
```

Reply:
@jcaamano Thanks for reviewing, I made it a separate function, ptal.

Reply:
Thanks, it does look better.

However, I was after keeping the non hardware offload path out of the retry block just for clarity. Is there any reason it needs to be in the retry block that I am missing?

Also, I am not that well versed in offloads but I am guessing this would only work with the tc offload model which I guess it is all that we use/support. But wouldn't work for example with DPDK offloads or, who knows, other models that might be introduced int he future.

Reply:
@jcaamano the function name `addOvsPortWithRetry` seems confusing, actually within the function, if hardware offload is not enabled, it will return at line 705 without retry. I can rename it to `addOvsPort` to avoid confusion.
Right now we focus on tc offload only, I can add a new function `verifyHWOffload`, so that dpdk or other future offload mechanism can add code to the function as well.

Reply:
@jcaamano right now in upstream we only support OVS Kernel datapath offloads using TC, so let us just please focus on this please. In future, when we add support for DPDK we will need to figure out how to verify HW offload.

Reply:
> @jcaamano right now in upstream we only support OVS Kernel datapath offloads using TC, so let us just please focus on this please. In future, when we add support for DPDK we will need to figure out how to verify HW offload.

That's fine, I just wanted to understand.

Reply:
> @jcaamano I'd like to avoid if-else block here. the function name `addOvsPortWithRetry` seems confusing, actually within the function, if hardware offload is not enabled, it will return at line 705 without retry. I can rename it to `addOvsPort` to avoid confusion. Right now we focus on tc offload only, I can add a new function `verifyHWOffload`, so that dpdk or other future offload mechanism can add code to the function as well.

I agree that the if-else block is not the prettiest of things. But I am confronted with the alternative of having to look into a sizable retry block that needs to check against `hwOffloadEnabled` twice for a code path that has nothing to do with it and wouldn't need any of that, which I would say is comparable in terms of ugliness at the very least. One has to really look into it to understand that there is no retries if there is no hw offload.

Anyway, this is relatively subjective, so I f you don't agree we can keep things as you have them now.

Reply:
@jcaamano I changed the function signature to reflect the retry requirement, please take a look, hope it address your concern.